common/djangoapps/third_party_auth/saml_configuration/tests/test_saml_configuration.py

```diff
@@ -0,0 +1,112 @@
+        'private_key': 'TestingKey',
+        'public_key': 'TestingKey',
+        'entity_id': 'example.com',
+        'is_public': False
+
+    },
+]
+
+TEST_PASSWORD = 'testpwd'
+
+@unittest.skipUnless(testutil.AUTH_FEATURE_ENABLED, testutil.AUTH_FEATURES_KEY + ' not enabled')
+class SAMLConfigurationTests(APITestCase):
+    """
+    API Tests for SAMLConfiguration objects retrieval.
+    The skip annotation above exists because we currently cannot run this test in
+    the cms mode in CI builds, where the third_party_auth application is not loaded
+    """
+    @classmethod
+    def setUpTestData(cls):
+        super(SAMLConfigurationTests, cls).setUpTestData()
+        cls.user = User.objects.create_user(username='testuser', password=TEST_PASSWORD)
+        cls.site, _ = Site.objects.get_or_create(domain='example.com')
+        for config in SAML_CONFIGURATIONS:
+            cls.samlconfiguration = SAMLConfiguration.objects.get_or_create(
+                site=cls.site,
+                slug=config['slug'],
+                private_key=config['private_key'],
+                public_key=config['public_key'],
+                entity_id=config['entity_id'],
+                is_public=config['is_public']
+            )
+        for config in PRIV_CONFIGURATIONS:
+            cls.samlconfiguration = SAMLConfiguration.objects.get_or_create(
+                site=cls.site,
+                slug=config['slug'],
+                private_key=config['private_key'],
+                public_key=config['public_key'],
+                entity_id=config['entity_id'],
+                is_public=config['is_public']
+            )
+
+
+    def setUp(self):
+        self.client.login(username=self.user.username, password=TEST_PASSWORD)
+
+    def test_get_saml_configurations_successful(self):
+        url = reverse('saml_configuration-list')
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # We ultimately just need ids and slugs, so let's just check those.
+        results = response.data['results']
+        self.assertEqual(results[0]['id'], SAML_CONFIGURATIONS[0]['site'])
+        self.assertEqual(results[0]['slug'], SAML_CONFIGURATIONS[0]['slug'])
+        self.assertEqual(results[1]['id'], SAML_CONFIGURATIONS[1]['site'])
+        self.assertEqual(results[1]['slug'], SAML_CONFIGURATIONS[1]['slug'])
+
+    def test_get_saml_configurations_noprivate(self):
+        # Verify we have 3 saml configuration objects: 2 public, 1 private.
+        total_object_count = SAMLConfiguration.objects.count()
+        self.assertEqual(total_object_count, 3)
+
+        url = reverse('saml_configuration-list')
+        response = self.client.get(url, format='json')
+
+        # We should only see 2 results, since 1 out of 3 are private
+        # and our queryset only returns public configurations.
+        results = response.data['results']
+        self.assertEqual(len(results), 2)
```

Comment:
optional: it may be great to also add id checks right here to ensure we are getting the right ids back? that should work if all ids are distinct

Reply:
Hmmm on second look, I'm already checking the ids as part of the test above it. I can check it again, but I've already tested the 2 public ones are part of the results set. Here I'm just also confirming we have 3 total objects (thus confirming the private did get created and is explicitly excluded.)